lib/logstash/outputs/rabbitmq.rb

```diff
@@ -44,22 +46,24 @@ class RabbitMQ < LogStash::Outputs::Base
       def register
         connect!
         @hare_info.exchange = declare_exchange!(@hare_info.channel, @exchange, @exchange_type, @durable)
-        @codec.on_event(&method(:publish))
+        # The connection close should close all channels, so it is safe to store thread locals here without closing them
+        @thread_local_channel = java.lang.ThreadLocal.new
+        @thread_local_exchange = java.lang.ThreadLocal.new
       end
 
       def symbolize(myhash)
         Hash[myhash.map{|(k,v)| [k.to_sym,v]}]
       end
 
-      def receive(event)
-        @codec.encode(event)
-      rescue StandardError => e
-        @logger.warn("Error encoding event", :exception => e, :event => event)
+      def multi_receive_encoded(events_and_data)
+        events_and_data.each do |event, data|
+          publish(event, data)
+        end
       end
 
       def publish(event, message)
         raise ArgumentError, "No exchange set in HareInfo!!!" unless @hare_info.exchange
```

Comment:
Is this ArgumentError still needed, now that hare_info.exchange is not used?

Reply:
Hmmmm, it's sort of a mystery to me why we have this in the first place. `declare_exchange` in `#register` should either raise an error or return an exchange. I'm not sure what this does.

I think this is beyond the scope of this PR given that.